Fix #75: non-default stations not matching

krisgesling opened the pull request:
Fix #75 

Had a quick look, it's not matching with high enough confidence so returns the default station.

Tidied up a little more, and it's passing VK and the old test runner. 
"Play the BBC News" is already included as a test phrase, however it's currently only testing that the Skill handles it. We'll need to go deeper to ensure the right feed is extracted and returned.

krisgesling committed:
add low confidence matching at title level

forslund commented on __init__.py:
```diff
@@ -161,28 +161,36 @@ def initialize(self):
-
-        if self.voc_match(phrase, "News"):
-            return ("news", CPSMatchLevel.TITLE)
+            match_feed_name(phrase, name, self.alt_feed_names[name])
```
Should the function return the name and confidence instead of passing through a shared variable?

Making it something like
```python
        def match_feed_name(phrase, source, alt_feed=None):                     
            phrase, feed_name = phrase.lower(), source.lower()                  
            # Test with News added in case user only says acronym eg "ABC".     
            # As it is short it may not provide a high match confidence.        
            conf = max(fuzzy_match(phrase, feed_name),                          
                    fuzzy_match(phrase, feed_name + self.translate("News")))
           return alt_feed or source, conf
                                                                  
        # Check primary feed list for matches eg 'ABC'                          
        for source in FEEDS:                                                    
            name, conf = match_feed_name(phrase, source)
            if conf > matched_feed['conf']:
                matched_feed['conf'] = conf
                matched_feed['name'] = name
```

krisgesling replied:
Good call, changed.

forslund reviewed (commented):
For me it is still not really working...mostly it uses the default station, however if I use `play news on bbc` npr-news is started.

krisgesling committed:
test that specific station chosen

krisgesling committed:
improve station searching

krisgesling commented:
I've also added another Step to the specific station tests to check that the right station is selected.
The searching now exclude's "the" from the phrase as this was returning too high a confidence for "other". 
Also added a search of the long name from the FEEDS dict which didn't exist before. This meant that a phrase like "ABC News Australia" would return a lower confidence than simply "ABC".

krisgesling committed:
refactor, renaming variables + docstring

forslund committed:
Add failing test cases

forslund commented:
It's much better :+1: but it doesn't quite cover the failing case in #75. I've added tests for a couple of these failing utterances.

Since this is much better I'm going to merge it but let #75 stay open.
